src/NHibernate/Transaction/AdoNetWithSystemTransactionFactory.cs

```diff
@@ -309,7 +309,7 @@ protected virtual void Unlock()
 			/// <param name="preparingEnlistment">The object for notifying the prepare phase outcome.</param>
 			public virtual void Prepare(PreparingEnlistment preparingEnlistment)
 			{
-				using (new SessionIdLoggingContext(_session.SessionId))
+				using (_session.BeginContext())
```

Comment:
Due to the possibility of having already closed session, it must here call only `BeginContext`, even if in some case and depending on the configuration, it may do some later operation requiring an opened session. (Those operations will still do the check themselves.)